app/script/user/UserRepository.js

```diff
@@ -0,0 +1,925 @@
+    return this._update_connection_status(user_et, z.user.ConnectionStatus.BLOCKED);
+  }
+
+  /**
+   * Cancel a connection request.
+   * @param {z.entity.User} user_et - User to cancel the sent connection request
+   * @param {z.entity.Conversation} next_conversation_et - Optional conversation to be switched to
+   * @returns {Promise} Promise that resolves when an outgoing connection request was cancelled
+   */
+  cancel_connection_request(user_et, next_conversation_et) {
+    return this._update_connection_status(user_et, z.user.ConnectionStatus.CANCELLED)
+      .then(function() {
+        if (next_conversation_et) {
+          amplify.publish(z.event.WebApp.CONVERSATION.SHOW, next_conversation_et);
+        }
+      });
+  }
+
+  /**
+   * Create a connection request.
+   * @param {z.entity.User} user_et - User to connect to
+   * @param {boolean} show_conversation - Should we open the new conversation
+   * @returns {Promise} Promise that resolves when the connection request was successfully created
+   */
+  create_connection(user_et, show_conversation = false) {
+    return this.user_service.create_connection(user_et.id, user_et.name())
+      .then((response) => {
+        return this.user_connection(response, show_conversation);
+      })
+      .catch((error) => {
+        return this.logger.error(`Failed to send connection request to user '${user_et.id}': ${error.message}`, error);
+      });
+  }
+
+  /**
+   * Get a connection for a user ID.
+   * @param {string} user_id - User ID
+   * @returns {z.entity.Connection} User connection entity
+   */
+  get_connection_by_user_id(user_id) {
+    for (const connection_et of this.connections()) {
+      if (connection_et.to === user_id) {
+        return connection_et;
+      }
+    }
+  }
+
+  /**
+   * Get a connection for a conversation ID.
+   * @param {string} conversation_id - Conversation ID
+   * @returns {z.entity.Connection} User connection entity
+   */
+  get_connection_by_conversation_id(conversation_id) {
+    for (const connection_et of this.connections()) {
+      if (connection_et.conversation_id === conversation_id) {
+        return connection_et;
+      }
+    }
+  }
+
+  /**
+   * Create a new conversation.
+   * @note Initially called by Wire for Web's app start to retrieve user entities and their connections.
+   * @param {number} limit - Query limit for user connections
+   * @param {string} user_id - User ID of the latest connection
+   * @param {Array<z.entity.Connection>} connection_ets - Unordered array of user connections
+   * @returns {Promise} Promise that resolves when all connections have been retrieved and mapped
+   */
+  get_connections(limit = 500, user_id, connection_ets = []) {
+    return this.user_service.get_own_connections(limit, user_id)
+      .then((response) => {
+        if (response.connections.length) {
+          const new_connection_ets = this.connection_mapper.map_user_connections_from_json(response.connections);
```

Comment:
I did the destructuring but I want to keep away from `connection_ets = connection_ets.concat(this.connection_mapper.map_user_connections_from_json(connections))` because it's quite a long line.